modules/mono/editor/GodotTools/GodotTools.Core/StringExtensions.cs

```diff
@@ -37,18 +38,40 @@ public static string NormalizePath(this string path)
         public static bool IsAbsolutePath(this string path)
         {
             return path.StartsWith("/", StringComparison.Ordinal) ||
-                       path.StartsWith("\\", StringComparison.Ordinal) ||
-                       path.StartsWith(driveRoot, StringComparison.Ordinal);
+                   path.StartsWith("\\", StringComparison.Ordinal) ||
+                   path.StartsWith(driveRoot, StringComparison.Ordinal);
         }
 
         public static string CsvEscape(this string value, char delimiter = ',')
         {
-            bool hasSpecialChar = value.IndexOfAny(new char[] { '\"', '\n', '\r', delimiter }) != -1;
+            bool hasSpecialChar = value.IndexOfAny(new char[] {'\"', '\n', '\r', delimiter}) != -1;
 
             if (hasSpecialChar)
                 return "\"" + value.Replace("\"", "\"\"") + "\"";
 
             return value;
         }
+
+        public static string ToSafeDirName(this string dirName, bool allowDirSeparator)
+        {
+            var invalidChars = new List<string> {":", "*", "?", "\"", "<", ">", "|"};
+            
+            if (allowDirSeparator)
+            {
+                // Directory separators are allowed, but disallow ".." to avoid going up the filesystem
+                invalidChars.Add("..");
```

Comment:
What if people write *only* `..` with no slashes? Or try to access a file called `test..txt` locally, then access it with a path, potentially silently failing? Shouldn't `..` always be disallowed?

Reply:
This is a mirror from the C++ method `OS::get_safe_dir_name`. It's indeed not perfect, but for what it's used, it won't silently fail. In our case, we use it to name the data folder. If the app name is _"My..Game"_ then the folder will be named _"data_My--Game"_. Not ideal considering _"data_My..Game"_ would be fine, but it still works.